src/handlers.rs

```diff
@@ -21,9 +21,13 @@ fn test_url() {
   assert!(issue_list_url.scheme() == "https");
 }
 
-fn string_from_ptr(ptr: *const c_char) -> String {
-  let cstr = unsafe { CStr::from_ptr(ptr as *const i8) };
-  String::from(cstr.to_str().unwrap())
+// Help. Is there a way to do this without macros?
```

Comment:
I don't think so -- an &str expects to store a reference to it's underlying String, but `*const c_char` isn't and can't be used as such (a String is not just a pointer, but also stores `length`, `capacity`, `refcount` etc.).
This is the reason you have to construct a new String here. 
If you make `str_from_ptr` a function then the String goes out of scope at the end of the function.
(you could also return both, but I suspect that might also defeat the purpose of this convenience wrapper).